src/engraving/dom/measure.cpp

```diff
@@ -1745,18 +1745,19 @@ EngravingItem* Measure::drop(EditData& data)
                 }
             }
         } else {
-            // drop to first end barline
-            seg = findSegmentR(SegmentType::EndBarLine, ticks());
-            if (seg) {
-                for (EngravingItem* ee : seg->elist()) {
-                    if (ee) {
-                        ee->drop(data);
-                        break;
-                    }
+            seg = undoGetSegmentR(SegmentType::EndBarLine, ticks());
+            // if any staff lacks a barline, create one
+            for (size_t stIdx = 0; stIdx < score()->nstaves(); ++stIdx) {
+                BarLine* bl = toBarLine(seg->element(stIdx * VOICES));
```

Comment:
This generates a warning on MSVC, because there is already something called `bl` in the outer scope